spendsterIOS/Presentation/HomeScreen/Activity/ActivityViewController.swift

```diff
@@ -15,41 +15,86 @@ class ActivityViewController: UIViewController, UITableViewDelegate, UITableView
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return (self.presenter?.expenses().count)!
-        //TODO
+        // TODO
         // real number of rows in server impl.
         // https://trello.com/c/uDMhmjnS/154-connect-expenses-to-the-server
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = self.expensesTableView.dequeueReusableCell(withIdentifier: "ExpenseItem", for: indexPath) as! TableViewCell
+        let cell = tableView.dequeueReusableCell(withIdentifier: "ExpenseItem", for: indexPath) as! ExpenseViewCell
         let expense = self.presenter?.expense(forIndex: indexPath.row)
         if let price = expense?.expenseAmount().description,
             let note = expense?.expenseNote() {
-            cell.bind(category: "Moked Category", image: "calendar", price: price, note: note)
+            cell.bind(category: "Moked Category", image: "oval", price: price, note: note)
         }
         return cell
-        //TODO
-        //real data binding in server impl.
+        // TODO
+        // real data binding in server impl.
         // https://trello.com/c/uDMhmjnS/154-connect-expenses-to-the-server
     }
     
+    
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        
+        if cell.responds(to: #selector(getter: UIView.tintColor)){
+            if tableView == self.expensesTableView {
+                let cornerRadius: CGFloat = 12.0
+                cell.backgroundColor = .clear
+                let layer: CAShapeLayer = CAShapeLayer()
+                let path: CGMutablePath = CGMutablePath()
+                let bounds: CGRect = cell.bounds
+                bounds.insetBy(dx: 25.0, dy: 0.0)
+                
+                if indexPath.row == 0 && indexPath.row == ( tableView.numberOfRows(inSection: indexPath.section) - 1) {
+                    path.addRoundedRect(in: bounds, cornerWidth: cornerRadius, cornerHeight: cornerRadius)
+                    
+                } else if indexPath.row == 0 {
```

Comment:
@Thongvor in such cases conditions can be extracted into functions with proper naming, describing the meaning